sample/rules.go

```diff
@@ -220,37 +253,44 @@ func ruleMatchesSpanInTrace(trace *types.Trace, rule *config.RulesBasedSamplerRu
 	return false
 }
 
-func extractValueFromSpan(trace *types.Trace, span *types.Span, condition *config.RulesBasedSamplerCondition, checkNestedFields bool) (interface{}, bool) {
+func extractValueFromSpan(trace *types.Trace, span *types.Span, condition *config.RulesBasedSamplerCondition, checkNestedFields bool) (value interface{}, exists bool, checkRootPrefix bool) {
+	checkedRootPrefix := false
```

Comment:
```suggestion
```
The specification of names in the return value declares those as local variables and initializes them to the zero value. With this initialization, you have created a new local variable that masks the return value. You should skip the initialization here.